Mathlib/MeasureTheory/Group/Measure.lean

```diff
@@ -246,6 +248,27 @@ end MeasurableMul
 
 end Mul
 
+section Semigroup
+
+variable [Semigroup G] [MeasurableMul G]
+
+/-- The image of a left invariant measure under right multiplication is left invariant.
+-/
+theorem isMulLeftInvariant_rmul [IsMulLeftInvariant μ] (g : G) :
+    IsMulLeftInvariant (map (· * g) μ) := by
+      let rmul_g := SMulCommClass.toMulActionHom G G (MulOpposite.op g)
+      have rmul_g_eq : ↑rmul_g = (· * g)
+      · ext x
+        simp only [SMulCommClass.toMulActionHom_apply, MulOpposite.smul_eq_mul_unop,
+          MulOpposite.unop_op]
+      have smul_inv := SMulInvariantMeasure_map μ rmul_g (measurable_mul_const g)
+      rw [rmul_g_eq] at smul_inv
+      rw [←forall_measure_preimage_mul_iff]
+      intro h s hS
+      exact smul_inv.measure_preimage_smul h hS
```

Comment:
Can you remove the `hS` from both the `intro` and `exact`? Sometimes these things cancel out.

Reply:
No :(